// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";
option optimize_for = LITE_RUNTIME;

import "components/manta/proto/common.proto";

package manta.proto;

// This forks google.rpc.Status.
message RpcStatus {
  int32 code = 1;
  string message = 2;
  repeated Proto3Any details = 3;
}

// This forks google.rpc.ErrorInfo.
message RpcErrorInfo {
  string reason = 1;
  string domain = 2;
  map<string, string> metadata = 3;
}

// This forks google.rpc.LocalizedMessage.
message RpcLocalizedMessage {
  // The locale used following the specification defined at
  // https://www.rfc-editor.org/rfc/bcp/bcp47.txt.
  // Examples are: "en-US", "fr-CH", "es-MX"
  string locale = 1;

  // The localized error message in the above locale.
  string message = 2;
}
